### ISSUE TITLE
mido.read_sysx_file() cannot read plain text hex

### ISSUE THREAD

**Issue by PaulForgey** [olemb/mido#80]
```
>>> m = mido.read_syx_file('test.txt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mido/syx.py", line 42, in read_syx_file
    data = bytearray.fromhex(data.decode('latin1'))
ValueError: non-hexadecimal number found in fromhex() arg at position 20
>>> 
[1]+  Stopped                 python
$ cat test.txt
F0 00 01 5D 02 00 F7
```

**Comment by olemb** [comment]
Ah, it turns out `bytearray.fromhex()` doesn't handle newlines:

```
>>> bytearray.fromhex('F0 00 01 5D 02 00 F7\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: non-hexadecimal number found in fromhex() arg at position 20
```

I also didn't use any newlines in my tests which means it has gone undetected since October.

Thanks for reporting! I'm working on a fix and will hopefully have it out sometime tomorrow. In the meantime here's a quick workaround:

```
import re
import mido

def read_syx_file(filename):
    with open(filename) as infile:
        text = infile.read()
        text = re.sub(r'\s', ' ', text)
        return mido.parse_all(bytearray.fromhex(text))
```

I'll let you know when the fix is out.

**Comment by olemb** [comment]
I just released 1.1.23 with a fix for this. Thanks again!
